### ISSUE TITLE
Where clause with DateTime doesn't work

### ISSUE THREAD

**Issue by yital9** [JetBrains/Exposed#972]
I'm using the last version of Exposed 0.26.1 with sqlite jdbc. 
Where clause doesn't seem to work properly, if some table has a DateTime column.

As an example I have a table `TestDates`
```
object TestDates : IntIdTable() {
    val time: Column<DateTime> = datetime("time").defaultExpression(CurrentDateTime())
}
class TestDate(id: EntityID<Int>) : IntEntity(id) {
    companion object : IntEntityClass<TestDate>(TestDates)
    var time by TestDates.time
}
```

The table seems to be created right:
_CREATE TABLE TestDates (id INTEGER PRIMARY KEY AUTOINCREMENT, time NUMERIC DEFAULT (CURRENT_TIMESTAMP) NOT NULL)_

Now I fill the table with test values:
```
for (i in 0..5) {
    Thread.sleep(10_000)
    transaction {
        TestDate.new {
        }
    }
}
```
It looks  correctly in the database browser:
1 | 2020-06-22 15:02:22
2 | 2020-06-22 15:02:32
3 | 2020-06-22 15:02:42
4 | 2020-06-22 15:02:52
5 | 2020-06-22 15:03:02
6 | 2020-06-22 15:03:12

Now I'm trying to make a simple query:
```
val count = transaction {
    addLogger(StdOutSqlLogger)
    TestDate.find { TestDates.time.greater(DateTime.parse("2020-06-22").withHourOfDay(15).withMinuteOfHour(2).withSecondOfMinute(50)) }.count()
}
```
and as a result I get `count == 6` from Exposed

But the query in log seems to be correct:
_SELECT COUNT(*) FROM TestDates WHERE TestDates.time > '2020-06-22 15:02:50.000000'_
and if I execute it manually I get the right answer `count == 3`

Am I doing something wrong or is it actually a bug?
Thank you in advance.

**Comment by Tapac** [comment]
Do you use jodatime or java-time?

**Comment by yital9** [comment]
I use jodatime

**Comment by hichem-fazai** [comment]
@yital9 I was trying to reproduce this bug using your code and I get count = 3 
I wanted to know if this bug still open or resolved ?

**Comment by yital9** [comment]
@hichem-fazai I can always reproduce the bug with 0.26.1 version. Interesting, that you don't. Can it be caused by local setup? Which other things could matter in this case?

**Comment by hichem-fazai** [comment]
@yital9 Are you sure that the loop `for (i in 0..5) { Thread.sleep(10_000)  ... }` gives you correct data? because for me it gives me the same value 6 times, without incrementing by 10 seconds. So, I had to insert the values manually like this :
````kotlin
TestDate.new { time = DateTime(2020,6,22,15,2,22) }
TestDate.new { time = DateTime(2020,6,22,15,2,32) }
TestDate.new { time = DateTime(2020,6,22,15,2,42) }
TestDate.new { time = DateTime(2020,6,22,15,2,52) }
TestDate.new { time = DateTime(2020,6,22,15,3,2) }
TestDate.new { time = DateTime(2020,6,22,15,3,12) }
````
And when calling your query I get count = 3

PS : For the setup of sqlite, I didn't do any extra configuration

**Comment by hichem-fazai** [comment]
@yital9 Ah now I can reproduce it ! When I use your loop `for (i in 0..5) { Thread.sleep(10_000) ... }` to insert data I get count = 6. 
At the begining I insert the values manually like below and I get count = 3 without any problem:

````Kotlin
TestDate.new { time = DateTime(2020,6,22,15,2,22) }
TestDate.new { time = DateTime(2020,6,22,15,2,32) }
TestDate.new { time = DateTime(2020,6,22,15,2,42) }
TestDate.new { time = DateTime(2020,6,22,15,2,52) }
TestDate.new { time = DateTime(2020,6,22,15,3,2) }
TestDate.new { time = DateTime(2020,6,22,15,3,12) }
````
I'm gonna make some investigation to understand the cause of this issue.

**Comment by yital9** [comment]
@hichem-fazai @Tapac 
I noticed a strange thing. 
If I set the value manually, as you did: 
`TestDate.new { time = DateTime(2020,6,22,15,2,42) }`
it is saved in the database as a `Long`
If it is set via `defaultExpression` as I did:
`TestDate.new {}`
it is represented as a `String`.

Might it be a problem?

I took a look into a source code of `DateColumnType` and is it possible, that the method
```
override fun notNullValueToDB(value: Any): Any = when {
        value is DateTime && time  -> java.sql.Timestamp(value.millis)
        value is DateTime -> java.sql.Date(value.millis)
        else -> value
}
```
has to convert values always into a String with `SQLITE_DATE_TIME_STRING_FORMATTER` and `SQLITE_DATE_STRING_FORMATTER`?

**Comment by hichem-fazai** [comment]
Yes indead, the problem occurs only when using `CURRENT_TIMESTAMP` to insert data,
I also tried to make the inserts using sqlite console but still having the same issue.
You're right, I guess that change must be done in `notNullValueToDB` function so that Exposed do some formatting when preparing the select statement.
I will give it a try today later, and make a PR

**Comment by hichem-fazai** [comment]
Yes indead, the problem occurs only when using `CURRENT_TIMESTAMP `to insert data,
I also tried to make the inserts using sqlite console but still having the same issue.
You're right, I guess that change must be done in `notNullValueToDB `function so that Exposed do some formatting when preparing the select statement.
I will give it a try today later, and make a PR if everything goes OK

**Comment by yital9** [comment]
Great, thank you! 
General question: which representation of a Date/Time is right? Long or String?
As I understood, you'd make a change for `CURRENT_TIMESTAMP ` so it will be also stored as a Long, right?
But wouldn't it be right to use always a String? 
According to sqlite spec it should be a ISO-8601 string: https://www.sqlite.org/lang_datefunc.html.
Actually I use the same database from python with Flask and it's a problem, that date/time fields are filled with Longs, because Flask can parse only String representation.

**Comment by hichem-fazai** [comment]
I think I will use string representation via `SQLITE_DATE_TIME_STRING_FORMATTER`
